### PR TITLE
Migration of Hub from DCL to Magic-Modules

### PR DESCRIPTION
**Pull request by KP08082132**
Migration of Hub from DCL to Magic-Modules
```release-note:note
networkconnectivity: `google_network_connectivity_hub` now uses mmv1 engine instead of DCL
```

### CODE REVIEW

**Review by c2thorn — COMMENTED** [review]
```
Error: google-beta/services/networkconnectivity/resource_network_connectivity_hub_test.go:28:29: undefined: testAccCheckNetworkConnectivityHubDestroyProducer
```
Without examples, the destroy producer function is not being generated. I'd recommend adding an example or copying the format using other `DestroyProducer` functions. Turning over to @slevenick for review

**Comment by slevenick** [comment]
> ```
> Error: google-beta/services/networkconnectivity/resource_network_connectivity_hub_test.go:28:29: undefined: testAccCheckNetworkConnectivityHubDestroyProducer
> ```
> 
> Without examples, the destroy producer function is not being generated. I'd recommend adding an example or copying the format using other `DestroyProducer` functions. Turning over to @slevenick for review

Yeah, we'll need a mmv1 example to be added so the generated tests work as intended, also for docs!

**Comment by KP08082132** [comment]
The legacy_long_form_project=true is not enabling the addition of projects/ before the project name, will that cause a breaking change?

**Comment by KP08082132** [comment]
MMv1 examples have been added for hub (basic and update), thanks!

**Comment by KP08082132** [comment]
"Field project lost its diff suppress function " should ideally be solved by adding legacy_long_form_projects to the top of the yaml, but it doesn't seem to be solving the issue

**Comment by KP08082132** [comment]
@slevenick @c2thorn please do take a look at the above

File: mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccNetworkConnectivityHub_BasicHubLongForm(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_network_connectivity_hub" "primary" {
+  name        = "long/form/tf-test-hub%{random_suffix}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go:87]:
```suggestion
  name        = "tf-test-hub%{random_suffix}"
```

File: mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  description = "A sample hub"
+  project     = "long/form/%{project_name}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go:89]:
```suggestion
  project     = "projects/%{project_name}"
```

File: mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+
+`, context)
+}
+
+func testAccNetworkConnectivityHub_BasicHubLongFormUpdate0(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_network_connectivity_hub" "primary" {
+  name        = "long/form/tf-test-hub%{random_suffix}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go:103]:
```suggestion
  name        = "tf-test-hub%{random_suffix}"
```

File: mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  description = "An updated sample hub"
+  project     = "long/form/%{project_name}"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go:105]:
```suggestion
  project     = "projects/%{project_name}"
```

Change this to projects/ as some things depend on it. These should fix the test failures

**Review by slevenick — COMMENTED** [review]
Tests are failing due to google-beta/services/networkconnectivity/resource_network_connectivity_hub_test.go:46:69: missing ',' in composite literal

File: mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "project"},
+			},
+			{
+				Config: testAccNetworkConnectivityHub_BasicHubLongFormUpdate0(context),
+			},
+			{
+				ResourceName:            "google_network_connectivity_hub.primary",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels, "project"},
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_hub_test.go:44]:
```suggestion
				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "project"},
```

**Comment by slevenick** [comment]
Adding @trodge as reviewer as I'm going to be out

**Comment by KP08082132** [comment]
@trodge please do take a look at the above, thanks!

**Review by trodge — APPROVED** [review]

**Pull request merged**